doc/stormond/stormond-hld.md

```diff
@@ -0,0 +1,263 @@
+
+The goal of the Storage Monitoring Daemon (stormond) is to provide meaningful metrics for the aforementioned issues and enable streaming telemetry for these attributes so that the required preventative measures are triggered in the eventuality of performance degradation.
+
+## 2. Data Collection
+
+We are intrested in the following characteristics that describe various aspects of the SSD:
+
+### **2.1 Priority 0 Attributes** 
+
+**The following are dynamic fields, offering up-to-date information that describes the current state of the SSD**
+
+    - IO Reads
+    - IO Writes
+    - Reserved Blocks Count
+    - Temperature
+
+**IO Reads/Writes** - SSDs use wear-leveling algorithms to distribute write and erase cycles evenly across the NAND cells to extend their lifespan. However, write amplification can occur when data is written, rewritten, and erased in a way that creates additional write operations, which can slow down performance.
+
+**Reserved Blocks Count** - Reserved blocks in a Solid State Drive (SSD) serve several critical purposes to enhance the drive's performance, reliability, and longevity. These reserved blocks are managed by the drive's firmware, and their specific allocation and management may vary between SSD manufacturers. The primary purposes of reserved blocks in an SSD are:
+
+- **Bad-block replacement:** When the firmware detects a bad block, it can map it to a reserved block and continue using the drive without data loss.
+- **Wear Leveling:** Reserved blocks are used to replace or relocate data from cells that have been heavily used, ensuring that all cells are used evenly. 
+- **Over-Provisioning:** Over-provisioning helps maintain consistent performance and extends the lifespan of the SSD by providing additional resources for wear leveling and bad block management.
+- **Garbage collection:** When files are deleted or modified, the old data needs to be erased and marked as available for new data. Reserved blocks can help facilitate this process by providing a temporary location to move valid data from blocks that need to be erased. 
+
+**Temperature** - Extreme temperatures can affect SSD performance. Excessive heat can lead to throttling to prevent damage, while extreme cold can slow down data access.
+
+
+### **2.2 Priority 1 Attributes**
+
+**These are a combination of static (S) and dynamic (D) fields, offering secondary information that provides additional context about the SSD**
+
+    - Vendor Model (S)
+    - Serial Number (S)
+    - Firmware (S)
+    - Health (D)
+
+These fields are self-explanatory.
+
+
+### **2.3 `stormond` Daemon Flow**
+
+1. `stormond` would be started by the `pmon` docker container
+2. The daemon would gather the static info once init-ed, by leveraging the `SsdUtil` class and update the StateDB
```

Comment:
@assrinivasan I guess IO R/W stats is not provided by ssdutil. if so please specify how those stat are collected?

Reply:
You are correct -- there is a sister-PR to stormond PR in sonic-platform-common that would enhance the functionality of SsdUtil to provide these stats. I will update the document to reflect this.

Reply:
Done in latest commit